### PR TITLE
[ClusterLoader2] change slice related Go template functions

### PR DESCRIPTION
**Pull request by jprzychodzen**

### CODE REVIEW

**Comment by mm4tt** [comment]
Could you provide some examples of how the new template functions should be used? And what is the problem with Seq you're trying to address?

File: clusterloader2/pkg/config/template_functions.go [hunk 1]
```diff
@@ -55,14 +57,18 @@ func GetFuncs() template.FuncMap {
...
 		"Seq":           seq,
+		"SliceOfZeros":  sliceOfZeros,
 		"SubtractFloat": subtractFloat,
 		"SubtractInt":   subtractInt,
 		"YamlQuote":     yamlQuote,
 	}
 }
 
+// seq returns a slice of size 'size' filled with zeros.
+// Deprecated: Naming generates confusion. Please use 'SliceOfZeros' for explicit zero values or 'Loop' for.
```
Comment (mm4tt) [inline clusterloader2/pkg/config/template_functions.go]: ... or 'Loop' for. 

Isn't there missing anything?

**Comment by mm4tt** [comment]
/approve
/lgtm

/hold

Feel free to cancel the hold once you fix the comment.

**Comment by jprzychodzen** [comment]
Just for a record:
`Seq` requires Go syntax to provide incremental integers - it's an error prone approach. With a single value rage, a number of `0` values is generated, and this is counter intuitive.

Example from clusterloader2/testing/experimental/storage/pod-startup/volume-types/downwardapi/deployment_with_downwardapi.yaml
Old syntax:
```yaml
{{ range $volumeIndex, $vol := Seq .VolumesPerPod }}
- name: vol-{{$volumeIndex}}
  mountPath: /usr/share/{{$volumeIndex}}
{{ end }}
```
`$vol` is discarded as not used

New syntax:
```yaml
{{ range $volumeIndex := Loop .VolumesPerPod }}
- name: vol-{{$volumeIndex}}
  mountPath: /usr/share/{{$volumeIndex}}
{{ end }}
```

File: clusterloader2/pkg/config/template_functions.go (hunk 1 above)
Comment (jprzychodzen) [reply clusterloader2/pkg/config/template_functions.go]: It was missing.

**Pull request merged**